code/modules/materials/Mat_ProcsDefines.dm

```diff
@@ -1,5 +1,8 @@
-
-var/global/list/material_cache
+// THESE TWO ARE THE SAME LIST
+/// the material cache, indexed by material id
+var/global/list/material_cache_matid
```

Comment:
why both? It seems like it'd be better to just have it be by type

Reply:
there's cases where i need it to still use id (for now)
just being safe. Ideally i'll remove it later

Reply:
okay i've decided to rip out the string matids at the cost of my sanity. I hope you are happy.